src/main/java/TaskList.java

```diff
@@ -0,0 +1,109 @@
+import javax.lang.model.type.ArrayType;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class TaskList {
+    private final ArrayList<Task> taskList;
+
+    public TaskList() {
+        taskList = new ArrayList<>();
+    }
+
+    private void add(Task task) {
+        this.taskList.add(task);
+    }
+
+    /**
+     * Called by Storage when Duke is initialised. Looks for the on hard disc storage of the list. If found, retrieves it to be used by Duke
+     * Else the text file is created to be used for future storing on hard disc
+     * @param File
+     * @return
+     * @throws IOException
+     */
+    public static TaskList retrieveTaskList(Path File) throws IOException {
+        TaskList returnTaskList = new TaskList();
+        if (java.nio.file.Files.exists(File)) {
+            Scanner scanner = new Scanner(File);
+            while (scanner.hasNextLine()) {
+                String taskData = scanner.nextLine();
+                String[] taskDataDivided = taskData.split(" ~ ");
+                boolean isDone = taskDataDivided[2].equals("1");
+                switch (taskDataDivided[0]) {
+                    case "E":
+                        returnTaskList.add(new Event(taskDataDivided[2], taskDataDivided[3], isDone));
```

Comment:
Should taskDataDivided[0, 2, 3] be assigned to a variable that is self-descriptive? It would make the code easier to read and understand.